src/controllers/confirm.sic.code.controller.ts

```diff
@@ -0,0 +1,16 @@
+import { NextFunction, Request, Response } from "express";
+import { getUrlWithCompanyNumber } from "../utils/url";
+import { TASK_LIST_PATH, urlParams } from "../types/page.urls";
+import { Templates } from "../types/template.paths";
+
+export const get = (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const companyNumber = req.params[urlParams.PARAM_COMPANY_NUMBER];
+    const backLinkUrl = getUrlWithCompanyNumber(TASK_LIST_PATH, companyNumber);
+    return res.render(Templates.SIC, {
+      backLinkUrl
+    });
+  } catch (e) {
+    return next(e);
```

Comment:
to test the exception happening, see if you can do something similar to this test that Jack did - https://github.com/companieshouse/confirmation-statement-web/blob/main/test/controllers/task.list.controller.unit.ts#L87

you could mock the getUrlWithCompanyNumber() function, and tell it to throw an error
`mockGetUrlWithCompanyNumber.mockImplementationOnce(() => {
  throw new Error();
});`

maybe add that mock implementation into your test function so it doesn't happen for the other tests.
Also, no need to check the mockCreateAndLogError in Jack's example as we don't use it here.
Give me a shout if you want a hand.